genkit-tools/common/src/types/model.ts

```diff
@@ -120,6 +120,8 @@ export const ModelInfoSchema = z.object({
       constrained: z.enum(['none', 'all', 'no-tools']).optional(),
       /** Model supports controlling tool choice, e.g. forced tool calling. */
       toolChoice: z.boolean().optional(),
+      /** Model can use an internal "thinking process" during response generation. */
+      thinkingMode: z.boolean().optional(),
```

Comment:
Let's exclude this from this PR. When we add first class support for thinking mode we'll do. I'm not sure if a simple boolean is going to sufficient. For example, see `constrained` option above, which needs to support other values.

```suggestion
```